net-nds/phpldapadmin/files/phpldapadmin-1.2.6.2-r1-default-templates.patch

```diff
@@ -0,0 +1,102 @@
+--- phpldapadmin/lib/PageRender.php_orig	2020-11-20 11:57:35.293875838 +0100
++++ phpldapadmin/lib/PageRender.php	2021-11-18 15:50:43.242593777 +0100
+@@ -360,6 +360,32 @@
+ 			else
+ 				$this->drawTemplateChoice();
+ 
++		# If mode is modification, use a DefaultForObjectClass template if exaclty 1 matches. Propose the template choice otherwise.
++		} elseif (($this->getMode() == 'modification') && (! isset($_REQUEST['template']))) {
++			if (DEBUGTMP) printf('<font size=-2>%s:<u>%s</u></font><br />',__METHOD__,'DEFAULTFOROBJECTCLASS template if only 1 template matches.');
++
++			$dn = $this->getModeContainer();
++			$server = $this->getServer();
++			$dnojcs = $server->getDNAttrValue($dn,'objectClass');
++			$matchingtmplIDs = array();
++			$unmatchingtmplIDs = array();
++			$tpldefojc = array();
++			foreach ($templates->getTemplates($this->getMode(),$this->getModeContainer(),true) as $tpl) {
++				$tpldefojc = $tpl->getDefaultForObjectClass();
++				if (array_intersect($tpldefojc,$dnojcs)) {
++					array_push($matchingtmplIDs,$tpl->getID());
++				} else {
++					array_push($unmatchingtmplIDs,$tpl->getID());
++				}
++			}
++			if (count($matchingtmplIDs) == 0)       # No matching DefaultForObjectClass found,
++				return 'none';                      # so we use the default template.
++			elseif (count($matchingtmplIDs) == 1)   # Exactly 1 matching DefaultForObjectClass found,
++				return $matchingtmplIDs[0];         # so we use it.
++			elseif (count($unmatchingtmplIDs) == 0) # No template found at all,
++				return 'none';                      # so we use the default template.
++			else                                    # Multiple templates found,
++				$this->drawTemplateChoice();        # so propose the template choice.
+ 		} else {
+ 			if (DEBUGTMP) printf('<font size=-2>%s:<u>%s</u></font><br />',__METHOD__,'SELECT a template to use.');
+ 
+--- phpldapadmin/lib/Template.php_orig	2020-11-20 11:57:35.276875606 +0100
++++ phpldapadmin/lib/Template.php	2021-11-18 15:50:05.381355470 +0100
```

Comment:
Usually we replace the file names with `a/lib/Template.php` and `b/lib/Template.php`, and also remove the extra info (timestamps are irrelevant).

You can use `scrub-patch` from `app-portage/iwdevtools`, it helps.

Reply:
I changed filenames to the "a b" form without extra info.